Zephyr fails to build with CPU load measurement enabled

**Describe the bug**
Cannot build with `TRACING_CPU_STATS` enabled as `smp_init` is both external linkage from kernel and internal in ble subsystem.

**To Reproduce**
Steps to reproduce the behavior:
1. Find some bluetooth enabled sample
2. enable `TRACING_CPU_STATS`
3. build

**Expected behavior**
should just work

**Impact**
annoyance, debug tool is unusable

**Screenshots or console output**
```
/home/pdunaj/work/ncs/zephyr/subsys/bluetooth/host/smp.c:2278:12: error: conflicting types for 'smp_init'
 static int smp_init(struct bt_smp *smp)
            ^~~~~~~~
In file included from /home/pdunaj/work/ncs/zephyr/arch/arm/include/kernel_arch_data.h:34,
                 from /home/pdunaj/work/ncs/zephyr/kernel/include/kernel_structs.h:68,
                 from /home/pdunaj/work/ncs/zephyr/subsys/debug/tracing/include/tracing_cpu_stats.h:10,
                 from /home/pdunaj/work/ncs/zephyr/include/tracing.h:34,
                 from /home/pdunaj/work/ncs/zephyr/include/kernel.h:4890,
                 from /home/pdunaj/work/ncs/zephyr/include/zephyr.h:18,
                 from /home/pdunaj/work/ncs/zephyr/subsys/bluetooth/host/smp.c:13:
/home/pdunaj/work/ncs/zephyr/kernel/include/kernel_internal.h:238:13: note: previous declaration of 'smp_init' was here
 extern void smp_init(void);
             ^~~~~~~~
```

**Environment (please complete the following information):**
master

**Additional context**
N/A

--- a/kernel/include/kernel_internal.h
+++ b/kernel/include/kernel_internal.h
@@ -235,7 +235,7 @@ extern void z_thread_monitor_exit(struct k_thread *thread);
 	} while (false)
 #endif /* CONFIG_THREAD_MONITOR */
 
-extern void smp_init(void);
+extern void z_smp_init(void);
 
 extern void smp_timer_init(void);
 

--- a/kernel/init.c
+++ b/kernel/init.c
@@ -263,7 +263,7 @@ static void bg_thread_main(void *unused1, void *unused2, void *unused3)
 	z_init_static_threads();
 
 #ifdef CONFIG_SMP
-	smp_init();
+	z_smp_init();
 #endif
 
 #ifdef CONFIG_BOOT_TIME_MEASUREMENT

--- a/kernel/smp.c
+++ b/kernel/smp.c
@@ -89,7 +89,7 @@ static void smp_init_top(int key, void *arg)
 }
 #endif
 
-void smp_init(void)
+void z_smp_init(void)
 {
 	(void)atomic_clear(&start_flag);
 

